### PR TITLE
[BugFix] Fix incorrect estimation of average row size when doing partial update

### PR DESCRIPTION
**Pull request by sevev**
When we do partial update, we first write a partial rowset and we will load the left columns data into memory during the following apply phase and we process the rowset data by segment. So if one segment file has too many rows, a large memory is needed. So we will get the average row size of historical data to avoid large memory alloc.

The following code is how we get the average row size. However, when we do partial update, the `data_size` is the compressed partial rowset data size which far less than total row size, so the `average_row_size` may far less than real average row size. So we may write too many rows into one segment which cause large memory alloc.
```
int64_t TabletUpdates::get_average_row_size() {
    TTabletInfo info;
    get_tablet_info_extra(&info);
    int64_t total_row = info.row_count;
    int64_t total_size = info.data_size;
    if (total_row != 0) {
        return total_size / total_row;
    } else {
        return 0;
    }
```
This pr fix the issue and return the correct average row size of historical data.

## What type of PR is this:
- [x] BugFix
- [ ] Feature
- [ ] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

## Checklist:
- [ ] I have added test cases for my bug fix or my new feature
- [ ] This pr will affect users' behaviors
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function

## Bugfix cherry-pick branch check:
- [x] I have checked the version labels which the pr will be auto-backported to the target branch
  - [x] 3.1
  - [x] 3.0
  - [x] 2.5
  - [ ] 2.4

### CODE REVIEW

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]


:heart_eyes: **pass** : 0 / 0 (0%)

File: be/src/storage/tablet_updates.cpp [hunk 1]
```diff
@@ -1280,7 +1286,11 @@ void TabletUpdates::_apply_normal_rowset_commit(const EditVersionInfo& version_i
         // 4. write meta
         const auto& rowset_meta_pb = rowset->rowset_meta()->get_meta_pb();
         if (rowset_meta_pb.has_txn_meta()) {
+            full_rowset_size = rowset->total_segment_data_size();
             rowset->rowset_meta()->clear_txn_meta();
+            rowset->rowset_meta()->set_total_row_size(full_row_size);
+            rowset->rowset_meta()->set_total_disk_size(full_rowset_size);
```
[the remark this answers is not in the document]
Comment (sevev) [reply be/src/storage/tablet_updates.cpp:1292]: > The `total_disk_size` is the same as `full_rowset_size`?

Yes, the `full_rowset_size` is the sum of all segment file size.

**Review by decster — APPROVED** [review]

**Review by chaoyli — APPROVED** [review]

**Comment by wanpengfei-git** [comment]
@Mergifyio backport branch-3.1

**Comment by wanpengfei-git** [comment]
@Mergifyio backport branch-3.0

**Comment by wanpengfei-git** [comment]
@Mergifyio backport branch-2.5

**Comment by yongbingwang** [comment]
branch-2.5
https://github.com/StarRocks/starrocks/pull/27862

**Comment by huoyue-git** [comment]
branch-3.0
https://github.com/StarRocks/starrocks/pull/27857

**Pull request merged**